node/_fs/_fs_streams.ts

```diff
@@ -1,55 +1,89 @@
+  mode?: number;
+  autoClose?: boolean;
+  emitClose?: boolean;
+  start?: number;
+  end?: number;
+  highWaterMark?: number;
+  fs?: Record<string, unknown> | null;
+}
 
-export class ReadStream extends NodeReadable {
-  public path: string;
+interface ReadStream extends NodeReadable {
+  path: string;
+}
 
-  constructor(path: string | URL, opts?: ReadStreamOptions) {
-    path = path instanceof URL ? fromFileUrl(path) : path;
-    const hasBadOptions = opts && (
-      opts.fd || opts.start || opts.end || opts.fs
+export function ReadStream(
+  this: ReadStream | unknown,
+  path: string | URL,
+  opts?: ReadStreamOptions & ReadableOptions,
+): ReadStream {
+  const hasBadOptions = opts && (
+    opts.fd || opts.start || opts.end || opts.fs
+  );
+  if (hasBadOptions) {
+    notImplemented(
+      `fs.ReadStream.prototype.constructor with unsupported options (${
+        JSON.stringify(opts)
+      })`,
     );
-    if (hasBadOptions) {
-      notImplemented(
-        `fs.ReadStream.prototype.constructor with unsupported options (${
-          JSON.stringify(opts)
-        })`,
-      );
-    }
-    const file = Deno.openSync(path, { read: true });
-    const buffer = new Uint8Array(16 * 1024);
-    super({
-      autoDestroy: true,
-      emitClose: true,
-      objectMode: false,
-      read: async function (_size) {
-        try {
-          const n = await file.read(buffer);
-          this.push(n ? Buffer.from(buffer.slice(0, n)) : null);
-        } catch (err) {
-          this.destroy(err as Error);
-        }
-      },
-      destroy: (err, cb) => {
-        try {
-          file.close();
-          // deno-lint-ignore no-empty
-        } catch {}
-        cb(err);
-      },
-    });
-    this.path = path;
   }
+
+  if (!(this instanceof ReadStream)) {
+    // deno-lint-ignore ban-ts-comment
+    // @ts-ignore
+    return new ReadStream(path, opts);
+  }
+
+  const _path = path instanceof URL ? fromFileUrl(path) : path;
+  const file = Deno.openSync(_path, { read: true });
+  const buffer = new Uint8Array(16 * 1024);
+
+  NodeReadable.call(this as ReadStream, {
+    highWaterMark: opts?.highWaterMark,
+    encoding: opts?.encoding,
+    objectMode: opts?.objectMode ?? false,
+    emitClose: opts?.emitClose ?? true,
+    autoDestroy: opts?.autoClose ?? opts?.autoDestroy ?? true,
+    signal: opts?.signal,
+    read(_size) {
+      try {
+        const n = file.readSync(buffer);
+        this.push(n ? Buffer.from(buffer.slice(0, n)) : null);
+      } catch (err) {
+        this.destroy(err as Error);
+      }
+    },
+    destroy(err, cb) {
+      try {
+        file.close();
+        // deno-lint-ignore no-empty
+      } catch {}
+      cb(err);
+    },
+  });
+
+  (this as ReadStream).path = _path;
+
+  return this as ReadStream;
 }
 
+Object.setPrototypeOf(ReadStream.prototype, NodeReadable.prototype);
+
 export function createReadStream(
   path: string | URL,
-  options?: ReadStreamOptions,
+  opts?: ReadStreamOptions,
```

Comment:
Why is this renamed? It looks like it is called `options` in the Node codebase and docs.

Reply:
The original `ReadStream`, `WriteStream`, and `createWriteStream` arguments were all `opts`.
It seemed odd that only the `createReadStream` argument was `options`, so I changed it.
I agree that `options` would be better, so I will change all of them.

Reply:
I have modified these in https://github.com/denoland/deno_std/pull/2634/commits/c12f3f1cd5a3d0039e991b86c952a5075ddecf8d